src/test/java/core/basesyntax/strategies/StrategiesTest.java

```diff
@@ -0,0 +1,139 @@
+package core.basesyntax.strategies;
+
+import static core.basesyntax.model.FruitTransaction.Operation.BALANCE;
+import static core.basesyntax.model.FruitTransaction.Operation.PURCHASE;
+import static core.basesyntax.model.FruitTransaction.Operation.RETURN;
+import static core.basesyntax.model.FruitTransaction.Operation.SUPPLY;
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.BalanceTransaction;
+import core.basesyntax.strategy.CalculationService;
+import core.basesyntax.strategy.CalculationStrategy;
+import core.basesyntax.strategy.PurchaseTransaction;
+import core.basesyntax.strategy.ReturnTransaction;
+import core.basesyntax.strategy.SupplyTransaction;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Test;
+
+public class StrategiesTest {
+    private static final String DEFAULT_FRUIT_APPLE = "apple";
+    private static final String DEFAULT_FRUIT_BANANA = "banana";
+    private final BalanceTransaction balanceTransaction = new BalanceTransaction();
+    private final PurchaseTransaction purchaseTransaction = new PurchaseTransaction();
+    private final ReturnTransaction returnTransaction = new ReturnTransaction();
+    private final SupplyTransaction supplyTransaction = new SupplyTransaction();
+    private final CalculationStrategy strategy = new CalculationStrategy();
+
+    @After
+    public void cleanStorageAfterEach() {
+        Storage.STORAGE.clear();
+    }
+
+    @Test
+    public void calculateAndStore_Balance_ok() {
+        Map<String, Integer> expected = new HashMap<>();
+        expected.put(DEFAULT_FRUIT_APPLE, 10);
+        expected.put(DEFAULT_FRUIT_BANANA, 35);
+        FruitTransaction transactionApple
+                = new FruitTransaction(BALANCE, DEFAULT_FRUIT_APPLE, 10);
+        FruitTransaction transactionBanana
+                = new FruitTransaction(BALANCE, DEFAULT_FRUIT_BANANA, 35);
```

Comment:
The same (creating instances, constants, tests for null and empty input parameters)